src/inputs/DateFields/utils.ts

```diff
@@ -0,0 +1,83 @@
+import { format as dateFnsFormat, isDate, parse as dateFnsParse } from "date-fns";
+import { DateRange } from "src/types";
+
+export type DateFieldMode = "single" | "range";
+
+export const dateFormats = {
+  short: "MM/dd/yy",
+  medium: "EEE, MMM d",
+  long: "EEEE LLLL d, uuuu",
+};
+
+export function getDateFormat(format: keyof typeof dateFormats | undefined) {
+  return format ? dateFormats[format] : dateFormats.short;
+}
+
+export function formatDate(date: Date | DateRange | undefined, format: string, mode: DateFieldMode) {
```

Comment:
Quibble: Not a huge fan of these taking `Date | DateRange` now...seems like ideally we could have separate `formatDate` and `formatDateRange` methods instead of moving/leaking the complexity of mode down these helper methods.

I don't have an immediate alternative of where to do it instead...

Reply:
I've updated to have separate functions specific for Date and DateRange, but it is kind of ugly at the moment as I need to do some casting:
```
(mode === "range"
      ? formatDateRange(value as DateRange | undefined, dateFormat)
      : formatDate(value as Date | undefined, dateFormat))
```

Once I get the typings correct, hopefully this will be cleaned up. Working on that now.